### DISCUSSION TITLE
how to 'refresh' note body when underlying template is changed

### DISCUSSION THREAD

**Discussion by jhub95** [TriliumNext/Trilium#3775] [Q&A]
Sorry if this question has been asked already. I spent some time looking through the discussions but didn't find what I was looking for.

How do we force a note body to update according the changes that have been made in it's connected template? For instance, the Books folder's children all use Book Template as a template, and include Main Characters, Plot, etc in the body of the note. When I change the template to include some more information, like link to Good Reads, none of the already existing book notes under Books update to include this new information in their body. Yes, promoted attributes are updated. What about the body? I understand this would be complicated as each note's body may have been changed in ways that a script would not know how to update it.

The downside is, that after having entered 100 books, then updating the body of the template, none of the 100 books will be updated with it. This makes me hesitant to commit to any one format for the body because I'll be stuck with it forever. Any advice is greatly appreciated. Cheers.

**Comment by zadam** [comment]
Hello, this is not possible. The template's content is really just the initial content and it's not automatically updated ever after.

It would be also basically impossible to merge the template content with the change you make (the content is free form HTML without any special markup which would assist in some form of merging).

**Comment by holoword** [comment]
As I guess, what you really need is something like database view other than template, is different top-down and bottom-up. In the top view can change all the time while you change the scheme, but inner content not change.
So, if you can write some js to autogenerate such page include your booknote page, I think it could achieve your goal.

**Comment by jhub95** [comment]
@zadam thanks for such a wonderful app. I was extremely glad I found it. Trillium is a high quality product.

In thinking through your answer it made me wonder if maybe we could use forms as a type of template. What if you included logic such that a note with #formtemplate then becomes an HTML form to which we add elements? 

To keep with the Books example, in the form template "BookFormTemplate" we could add `<textarea id="MainCharacters"></textarea>`. Then, we create a note called "Hobbit" and relate it to the template with `~formtemplate="BookFormTemplate"`. The note "Hobbit" is not a replication of the template, but rather there are divs which store the content of the form elements. For instance, inside of "Hobbit" would be `<div elementID="MainCharacters">Bilbo, <em>Gandalf</em></div>`

Then when the "Hobbit" note is loaded for viewing, the form elements found in BookFormTemplate are then loaded with the content from their respective DIVs: `<textarea id="MainCharacters><div elementID="MainCharacters">Bilbo, <em>Gandalf</em></div></textarea>`

Later on, when we decide to add another form element `<textarea id="plot">` to the form template, we will see the new form element even in the old note "Hobbit". Of course it will be blank because there is no corresponding div in the note to go with "plot".

Text area elements could include the code that allows formatting of text within the div that stores the content.

When a note is purposely disconnected from the template to become independent, at that point the form and the content could be merged into the note and left as HTML, never again loading the dynamic form elements.

Just brain storming.

**Reply by marco10x15** [reply]
This would be a great add on!

**Reply by mikkel1156** [reply]
I was missing a feature like this after trying out other software, and found an issue which is kind of what you are mentioning - made a simple MVP using tables (which is my use-case) https://github.com/zadam/trilium/issues/2655

However that's being tied up by working with attributes, but guessing it could be adapted to look through the content?
